src/AuthenticationServices/PublicPrivateKeyAuthentication.cs

```diff
@@ -0,0 +1,33 @@
+//
+// PublicPrivateKeyAuthentication.cs
+//
+// Authors:
+//	TJ Lambert <antlambe@microsoft.com>
+//
+// Copyright 2021 Microsoft Corporation
+//
+
+#if !TVOS && !WATCH
+
+using System;
+using System.Runtime.InteropServices;
+using CoreGraphics;
+using Foundation;
+using ObjCRuntime;
+
+#nullable enable
+
+namespace AuthenticationServices {
+
+    [NoWatch, NoTV, Mac (12,0), iOS (15,0), MacCatalyst (15,0)]
+	public static class PublicPrivateKeyAuthentication {
+        [DllImport (Constants.AuthenticationServicesLibrary)]
+        static extern /* NSString[] */ IntPtr ASAuthorizationAllSupportedPublicKeyCredentialDescriptorTransports ();
+
+        public static /* ASAuthorizationSecurityKeyPublicKeyCredentialDescriptorTransport[] */ NSString[] GetAllSupportedPublicKeyCredentialDescriptorTransports () {
+            return NSArray.ArrayFromHandle<NSString> (ASAuthorizationAllSupportedPublicKeyCredentialDescriptorTransports ());
+        }
```

Comment:
you have `ASAuthorizationSecurityKeyPublicKeyCredentialDescriptorTransport` defined
there's no point in exposing the list (only) as `NSString`

Reply:
Are you suggesting to do something like this?
```
public static ASAuthorizationSecurityKeyPublicKeyCredentialDescriptorTransport[] GetAllSupportedPublicKeyCredentialDescriptorTransports () {
	return NSArray.ArrayFromHandle<ASAuthorizationSecurityKeyPublicKeyCredentialDescriptorTransport> (ASAuthorizationAllSupportedPublicKeyCredentialDescriptorTransports ());
}
```

Reply:
For the API signature ? yes
Not sure the implementation will compile...

Reply:
it does not currently compile since ASAuthorizationSecurityKeyPublicKeyCredentialDescriptorTransport is an enum and it complains saying that it needs to be a reference type: 
```
error CS0452: The type 'ASAuthorizationSecurityKeyPublicKeyCredentialDescriptorTransport' must be a reference type in order to use it as parameter 'T' in the generic type or method 'NSArray.ArrayFromHandle<T>(IntPtr)'
```
Should I try turning the enum into an interface so that it can be a reference type?

Reply:
no, you just need a bit more code
You already get an array of `NSString` and you can create an array of the `enum` and convert each string constant into it's enum value